### ISSUE TITLE
Unknown OSGi execution environment: 'JavaSE-11

### ISSUE THREAD

**Issue by gorogoro2** [COVESA/capicxx-core-tools#46]
Hi,

I'm havong trouble when compiling this.

I got the following error:

```java
[ERROR] Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/miguel/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11' -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/miguel/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11'
   at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120 undefined)
   at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957 undefined)
   at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289 undefined)
   at org.apache.maven.cli.MavenCli.main (MavenCli.java:193 undefined)
   at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
   at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62 undefined)
   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43 undefined)
   at java.lang.reflect.Method.invoke (Method.java:498 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347 undefined)
Caused by: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/miguel/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11'
   at org.eclipse.tycho.core.osgitools.OsgiManifest.parseExecutionEnvironments (OsgiManifest.java:69 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiManifest.<init> (OsgiManifest.java:55 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiManifest.parse (OsgiManifest.java:152 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifestFromFile (DefaultBundleReader.java:82 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.doLoadManifest (DefaultBundleReader.java:60 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifest (DefaultBundleReader.java:49 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.loadManifest (EquinoxResolver.java:269 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.newState (EquinoxResolver.java:161 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.newResolvedState (EquinoxResolver.java:69 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:254 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:156 undefined)
   at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:130 undefined)
   at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:77 undefined)
   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264 undefined)
   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192 undefined)
   at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105 undefined)
   at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957 undefined)
   at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289 undefined)
   at org.apache.maven.cli.MavenCli.main (MavenCli.java:193 undefined)
   at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
   at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62 undefined)
   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43 undefined)
   at java.lang.reflect.Method.invoke (Method.java:498 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347 undefined)
Caused by: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-11'
   at org.eclipse.tycho.core.ee.ExecutionEnvironmentUtils.getExecutionEnvironment (ExecutionEnvironmentUtils.java:82 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiManifest.parseExecutionEnvironments (OsgiManifest.java:66 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiManifest.<init> (OsgiManifest.java:55 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiManifest.parse (OsgiManifest.java:152 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifestFromFile (DefaultBundleReader.java:82 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.doLoadManifest (DefaultBundleReader.java:60 undefined)
   at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifest (DefaultBundleReader.java:49 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.loadManifest (EquinoxResolver.java:269 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.newState (EquinoxResolver.java:161 undefined)
   at org.eclipse.tycho.core.osgitools.EquinoxResolver.newResolvedState (EquinoxResolver.java:69 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:254 undefined)
   at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:156 undefined)
   at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:130 undefined)
   at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:77 undefined)
   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264 undefined)
   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192 undefined)
   at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105 undefined)
   at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957 undefined)
   at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289 undefined)
   at org.apache.maven.cli.MavenCli.main (MavenCli.java:193 undefined)
   at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
   at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62 undefined)
   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43 undefined)
   at java.lang.reflect.Method.invoke (Method.java:498 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406 undefined)
   at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347 undefined)
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException

```

I have the java 8 version:
```shell
$ java -version                                      
openjdk version "1.8.0_352"
OpenJDK Runtime Environment (build 1.8.0_352-8u352-ga-1~22.04-b08)
OpenJDK 64-Bit Server VM (build 25.352-b08, mixed mode)                  
```

It also happens with java 11 though.

Am I missing something? Why is it complaining of java 11 but when I do `java -version` it shows me that I'm using Java8?
I'm compiling this on UBUTU WSL system.

Thanks

**Comment by freekevinCode** [comment]
Instead of openjdk, try oracle jdk11，

**Comment by congtaowang** [comment]
MY JRE

java 11.0.18 2023-01-17 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.18+9-LTS-195)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.18+9-LTS-195, mixed mode)

BUT

Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/zhuima/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11' -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/zhuima/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11'
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST /home/zhuima/.m2/repository/p2/osgi/bundle/org.eclipse.emf.mwe.core/1.8.0.v20221117-1134/org.eclipse.emf.mwe.core-1.8.0.v20221117-1134.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-11'
    at org.eclipse.tycho.core.osgitools.OsgiManifest.parseExecutionEnvironments (OsgiManifest.java:69)
    at org.eclipse.tycho.core.osgitools.OsgiManifest.<init> (OsgiManifest.java:55)
    at org.eclipse.tycho.core.osgitools.OsgiManifest.parse (OsgiManifest.java:152)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifestFromFile (DefaultBundleReader.java:82)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.doLoadManifest (DefaultBundleReader.java:60)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifest (DefaultBundleReader.java:49)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.loadManifest (EquinoxResolver.java:269)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.newState (EquinoxResolver.java:161)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.newResolvedState (EquinoxResolver.java:69)
    at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:254)
    at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:156)
    at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:130)
    at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:77)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.eclipse.tycho.core.ee.UnknownEnvironmentException: Unknown OSGi execution environment: 'JavaSE-11'
    at org.eclipse.tycho.core.ee.ExecutionEnvironmentUtils.getExecutionEnvironment (ExecutionEnvironmentUtils.java:82)
    at org.eclipse.tycho.core.osgitools.OsgiManifest.parseExecutionEnvironments (OsgiManifest.java:66)
    at org.eclipse.tycho.core.osgitools.OsgiManifest.<init> (OsgiManifest.java:55)
    at org.eclipse.tycho.core.osgitools.OsgiManifest.parse (OsgiManifest.java:152)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifestFromFile (DefaultBundleReader.java:82)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.doLoadManifest (DefaultBundleReader.java:60)
    at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifest (DefaultBundleReader.java:49)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.loadManifest (EquinoxResolver.java:269)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.newState (EquinoxResolver.java:161)
    at org.eclipse.tycho.core.osgitools.EquinoxResolver.newResolvedState (EquinoxResolver.java:69)
    at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:254)
    at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:156)
    at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:130)
    at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:77)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException

ALSO

**Comment by jiangchaochao** [comment]
解决了吗，我也碰到这个问题

**Comment by Gustavo1515** [comment]
Issue fixed with the PR #50 

Cc: @goncaloalmeida
